lightning/src/routing/network_graph.rs

```diff
@@ -1170,6 +1190,74 @@ impl NetworkGraph {
 		self.update_channel_intern(msg, None, None::<(&secp256k1::Signature, &Secp256k1<secp256k1::VerifyOnly>)>)
 	}
 
+	/// Override a (directed) channel's individual properties without having a full update
+	/// This method should be used exclusively in conjunction with rapid graph sync, which is intended
+	/// for mobile devices.
+	///
+	/// `short_channel_id`: Channel to be updated
+	///
+	/// `flags`: Channel flags to set. Least significant bit serves as directional indicator
+	///
+	/// `mutations: Vec<ChannelMutationProperty>`: Array of properties to mutate
+	pub fn override_directional_channel_properties(&self, short_channel_id: u64, flags: u8, timestamp: u32, chain_hash: BlockHash, mutations: ChannelMutationProperties) -> Result<(), LightningError> {
+		let channel_update = {
+			let mut channels = self.channels.read().unwrap();
+			let direction = flags & 1u8;
+			match channels.get(&short_channel_id) {
+				None => return Err(LightningError{err: "Couldn't find channel for update".to_owned(), action: ErrorAction::IgnoreError}),
+				Some(channel) => {
+					let directional_info = if direction == 0 {
+						channel.one_to_two.as_ref()
+					} else {
+						channel.two_to_one.as_ref()
+					};
+					if let Some(directional_info) = directional_info {
+
+						let htlc_maximum_msat = if let Some(htlc_maximum_msat) = directional_info.htlc_maximum_msat {
+							OptionalField::Present(htlc_maximum_msat)
+						} else {
+							OptionalField::Absent
+						};
+
+						let mut synthetic_update = msgs::UnsignedChannelUpdate {
+							chain_hash,
+							short_channel_id,
+							timestamp,
+							flags,
+							cltv_expiry_delta: directional_info.cltv_expiry_delta,
+							htlc_minimum_msat: directional_info.htlc_minimum_msat,
+							htlc_maximum_msat,
+							fee_base_msat: directional_info.fees.base_msat,
+							fee_proportional_millionths: directional_info.fees.proportional_millionths,
+							excess_data: vec![]
+						};
```

Comment:
The `UnsignedChannelUpdate` for the full update is constructed in `lightning-graph-sync` while the incremental update is formed here. For consistency and to avoid blurring layers, could we do both in `lightning-graph-sync`? (Related: see next comment about locking.)

Then you can make a `CustomChannelUpdate` enum in `lightning-graph-sync` with variants for full and incremental with and implementation of `Readable` there and possibly conversion functions to `UnsignedChannelUpdate`.

Reply:
I agree, and I would have loved to pull the incremental update out of the lightning crate. The reason I haven't is because we need access to the previous directional channel info, and that for sure seems like a property that should not be made public. Very open to suggestions. 

Additionally, conversion functions don't really work because it it is reliant on the previous directional channel info.

Reply:
> I agree, and I would have loved to pull the incremental update out of the lightning crate. The reason I haven't is because we need access to the previous directional channel info, and that for sure seems like a property that should not be made public. Very open to suggestions.
> 

Isn't the directional info already part of `NetworkGraph`'s public interface? We even have `ChannelInfo::as_direct_to`.

Reply:
I think I have found a way to do that.

Reply:
done